### ISSUE TITLE
Output of DeepExplainer

### ISSUE THREAD

**Issue by guyj-s1** [shap/shap#2204]
Hi,
I'm using DeepExplainer with a PyTorch model that outputs probabilities (softmax) for a binary classification task.
I'm wondering about the shap_values, whether or not they are in "probability" or "logit" units. 
Namely, for a sample to be explained, is each feature given a shap_value that is the amount of "probability" needed to get from the expected_value of the class to the model's output for that class? because those two are in "probability space".

I'm asking because: sum(shap_values) + expected_value != model_output (for a given class)

Help would be much appreciated!

Thanks,
Guy

**Comment by saifullah3396** [comment]
I am also facing this on every model I try except for the mnist example given in the docs. I tried Alexnet with th same mnist example and it doesn't seem to satisfy this equation. Although the shap values still stay proportional to the output relatively. (Note: for any networks you're running, make sure relu inplace=False and no layer is repeated (e.g repitition of same relu activation) as these two conditions are not yet considered in shap deepexplainer computations as far as I could see). Also, avoid any kind of functional layer in the forward pass.

Also could you please post your output regarding how much it the equation deviates for each class?

**Comment by rashecl** [comment]
For me, sum(shap_values) + expected_value == model_output when using a shap.DeepExplainer on keras model on a binary classification task. And the shap values appear to be output on a probability scale. I would like to output shap values in a log-odds domain. Is that possible @slundberg?

**Comment by rabyj** [comment]
Thanks @saifullah3396, your comment helped me.

Turns out, in my case, the problem was that my model `forward` outputted probabilities, as so
```Python
def forward(self, x: torch.Tensor):
    """Return probabilities."""
    return F.softmax(self.forward_train(x), dim=1)

def forward_train(self, x):
    """Return logits."""
    return self._pt_model(x) # output of nn.Sequential(*layer_list)
```
By changing my model to output logits, I finally obtained `sum(shap_values) + expected_value == model_output`.

I think this answers the OP question, as `DeepExplainer` seems to expect "raw" output from the model, with no final functional layer like softmax in the forward, from what I understand. 

This issue could be closed now @guyj-s1 ?
